03.bowling/bowling.rb

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/ruby
+# frozen_string_literal: true
+
+class MyBowling
+  def initialize(text)
+    create_frames(text)
+  end
+
+  def result
+    total_score = 0
+    @frames.each_with_index do |frame, i|
+      total_score += frame.sum
+
+      # 10フレーム目はボーナスの加算がないので抜ける
+      break if i == 9
+
+      total_score += calc_strike_bonus(i) if strike?(frame)
+      total_score += calc_spare_bonus(i) if spare?(frame)
+    end
+    puts total_score
+  end
+
+  private
+
+  def strike?(frame)
+    frame[0] == 10
+  end
+
+  def spare?(frame)
+    !strike?(frame) && frame.sum == 10
+  end
+
+  def calc_strike_bonus(index)
+    # 次のフレームもストライクの場合は、次の次のフレームの1投目のスコアも加算する
+    bonus = 0
+    next_frame = @frames[index + 1]
+    next_next_frame = @frames[index + 2]
+    if next_frame[0] == 10
+      bonus += 10
+      # 9フレーム目だけ10フレーム目の2投目を加算する
+      bonus += index < 8 ? next_next_frame[0] : next_frame[1]
+    else
+      bonus += next_frame[0] + next_frame[1]
+    end
+    bonus
+  end
+
+  def calc_spare_bonus(index)
+    next_frame = @frames[index + 1]
+    next_frame[0]
```

Comment:
```suggestion
    @frames[index + 1][0]
```

ロジックらしいロジックがないのでこれで十分かも

Reply:
修正しました 🙏🏼 

[977483a](https://github.com/nagatanuen/ruby-practices/pull/3/commits/977483a76282e61bc3174f440c1fd7ab9c595a64)